### PR TITLE
Design doc for delete feature

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 
https://github.com/pingcap/tispark/issues/2235

### What is changed and how it works?
Add design doc

### CODE REVIEW

File: docs/design/2022-03-21-delete-support.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+## Motivation or Background
+
+TiSpark is a connector that provides read and write support. However, it lacks the ability to delete. 
+
+Delete operations are becoming more and more important for many big-data workflows. Right now, we are able to provide delete operations after the support of delete in spark 3.0.
+
+## Goals
+**Must have**
+- Support delete bypass TiDB and provide friendly API with spark SQL
+- Compatible with TiDB 4. x and 5. x
```
Comment (zhangyangyu) [inline docs/design/2022-03-21-delete-support.md:39]:
```suggestion
- Compatible with TiDB 4.x and 5.x 
```

File: docs/design/2022-03-21-delete-support.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+Delete operations are becoming more and more important for many big-data workflows. Right now, we are able to provide delete operations after the support of delete in spark 3.0.
+
+## Goals
+**Must have**
+- Support delete bypass TiDB and provide friendly API with spark SQL
+- Compatible with TiDB 4. x and 5. x 
+
+**Nice to have**
+- Optimize the code in the write feature which can be reused. Because write and delete have similar logic. However, the code in the write is hard to reuse.
```
Comment (zhangyangyu) [inline docs/design/2022-03-21-delete-support.md:42]: The code for write is hard to use **currently** so we need to optimize, right?

File: docs/design/2022-03-21-delete-support.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+- User doc
+
+**Not Support**
+- Delete without WHERE clause
+- Don't support delete statement with subQuery
+
+## API
+Use spark SQL to excute delete in TiSpark.
```
Comment (zhangyangyu) [inline docs/design/2022-03-21-delete-support.md:50]:
```suggestion
Use spark SQL to execute delete in TiSpark.
```

File: docs/design/2022-03-21-delete-support.md (hunk 2 above)
Comment (shiyuhang0) [reply docs/design/2022-03-21-delete-support.md:42]: yes

**Review by xuanyu66 — APPROVED** [review]

**Review by sunxiaoguang — APPROVED** [review]

**Pull request merged**